src/main/java/org/folio/service/transactions/BatchTransactionService.java

```diff
@@ -51,28 +94,40 @@ private void sanityChecks(Batch batch) {
 
   private void checkIdIsPresent(List<Transaction> transactions, String operation) {
     for (Transaction transaction : transactions) {
-      if (StringUtils.isBlank(transaction.getId())) {
+      if (isBlank(transaction.getId())) {
         throw new HttpException(400, String.format("Id is required in transactions to %s.", operation));
       }
+      if (TYPES_WITH_SUMMARIES.contains(transaction.getTransactionType()) && isBlank(transaction.getSourceInvoiceId()) &&
+          (transaction.getEncumbrance() == null || isBlank(transaction.getEncumbrance().getSourcePurchaseOrderId()))) {
+        throw new HttpException(400, String.format("Missing invoice id or order id in transaction %s", transaction.getId()));
+      }
     }
   }
 
-  private Future<Void> createTransactions(List<Transaction> transactions, DBConn conn) {
+  private Future<Void> createTransactions(List<Transaction> transactions, DBConn conn,
+      Map<String, String> okapiHeaders) {
     if (transactions.isEmpty())
       return Future.succeededFuture();
-    return Future.failedFuture(new HttpException(500, "transactionsToCreate: not implemented"));
+    populateMetadata(transactions, okapiHeaders);
+    return createOrUpdate(transactions,
+        (transactionService, transaction) -> transactionService.createTransaction(transaction, conn).mapEmpty(), conn)
+      .onSuccess(v -> logger.info("Batch transactions: successfully created transactions"))
```

Comment:
please add transactions ids or count in each log in this class, in this case it would be simpler to understand how many transactions are sent from mod-orders/mod-invoice that can be useful

Reply:
OK